internal/provider/provider.go

```diff
@@ -146,12 +158,16 @@ func (p *CiscoMerakiProvider) Configure(ctx context.Context, req provider.Config
 
 	// MERAKI BASE URL
 	if !data.BaseUrl.IsNull() {
+		merakiURL, err := url.JoinPath(data.BaseUrl.ValueString(), data.ApiVersion.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("unable to join meraki url ", err.Error())
+			return
+		}
 
-		baseUrl, err := url.Parse(data.BaseUrl.ValueString())
 		if err == nil {
 			configuration.Servers = openApiClient.ServerConfigurations{
```

Comment:
Reviewing the structure of serverConfigurations:

`// ServerVariable stores the information about a server variable
type ServerVariable struct {
	Description  string
	DefaultValue string
	EnumValues   []string
}

// ServerConfiguration stores the information about a server
type ServerConfiguration struct {
	URL string
	Description string
	Variables map[string]ServerVariable
}

// ServerConfigurations stores multiple ServerConfiguration items
type ServerConfigurations []ServerConfiguration`

We see how it's implemented by default with the HTTP client:

`// NewConfiguration returns a new Configuration object
func NewConfiguration() *Configuration {
	cfg := &Configuration{
		DefaultHeader:    make(map[string]string),
        // current = meraki-golang/v1.36.0
		UserAgent:   merakiUserAgent("v1.39.0", os.Getenv("MERAKI_GO_SDK_CALLER")),
		Debug:            false,
		Servers:          ServerConfigurations{
			{
				URL: "https://api.meraki.com/{basePath}",
				Description: "No description provided",
				Variables: map[string]ServerVariable{
					"basePath": ServerVariable{
						Description: "Meraki API Go Client",
						DefaultValue: "api/v1",
					},
				},
			},
		},
		OperationServers: map[string]ServerConfigurations{
		},
        DefaultBaseUrl: getEnvOrDefaultStr("DEFAULT_BASE_URL","https://api.meraki.com/api/v1"),
       // other config items ...
	}
	return cfg
}`

I recommend updating the regex to strongly enforce the schema (without the / since it's a HTTP primitive), and to use the schema Default parameter to set them if not specified to:

BaseUrl = "api.meraki.com"
ApiVersion = "api/v1"


From there we can pass them along to ServerConfigurations:

`
ServerConfigurations{
			{
				URL: data.BaseUrl.StringValue() + "/{basePath}",
				Description: "No description provided",
				Variables: map[string]ServerVariable{
					"basePath": ServerVariable{
						Description: "Meraki API Go Client",
						DefaultValue: data.ApiVersion.StringValue(),
					},
				},
			},
		},
`